src/DebtRequest.hs

```diff
@@ -93,16 +93,16 @@ mkValidator contractInfo@ContractInfo{..} dat borrowerTn ctx = validate
     findDatumHash' :: ToData a => a -> TxInfo -> Maybe DatumHash
     findDatumHash' datum info = findDatumHash (Datum $ toBuiltinData datum) info
 
-    expectedNewDatum :: POSIXTime -> Collateral.CollateralDatum
-    expectedNewDatum ld = Collateral.CollateralDatum {
+    expectedNewDatum :: POSIXTime -> Integer -> Collateral.CollateralDatum
+    expectedNewDatum ld updatedColat = Collateral.CollateralDatum {
         Collateral.borrowersNftTn        = borrowerTn
       , Collateral.borrowersAddress      = borrowersAddress dat
       , Collateral.loan                  = loan dat
       , Collateral.loanAmnt              = loanAmnt dat
       , Collateral.interest              = interest dat
       , Collateral.interestAmnt          = interestAmnt dat
       , Collateral.collateral            = collateral dat
-      , Collateral.collateralAmnt        = collateralAmnt dat
+      , Collateral.collateralAmnt        = updatedColat
       , Collateral.loanDuration          = loanDuration dat
       , Collateral.liquidateNft          = liquidateNft dat
       , Collateral.collateralFactor      = collateralFactor dat
```

Comment:
I think this needs to be updated as well, if the `collateralAmnt` can be changed